modules/history/src/main/HistoryApi.scala

```diff
@@ -9,13 +9,26 @@ import scala.concurrent.duration._
 import chess.Speed
 import lila.db.dsl._
 import lila.game.Game
-import lila.rating.PerfType
+import lila.rating.{ Perf, PerfType }
 import lila.user.{ User, Perfs }
 
 final class HistoryApi(coll: Coll) {
 
   import History._
 
+  def add_puzzle(user: User, completedAt: DateTime, perf: Perf): Funit = {
+    val days = daysBetween(user.createdAt, completedAt)
+    coll.update(
+      $id(user.id),
+      $doc(BSONDocument(
+        "$set" -> (
+          BSONDocument(s"puzzle.$days" -> $int(perf.intRating))
+        )
+      )),
```

Comment:
```
      $doc(BSONDocument(
        "$set" -> (
          BSONDocument(s"puzzle.$days" -> $int(perf.intRating))
        )
      ))
```
==
`$set(s"puzzle.$days" -> $int(perf.intRating))`

Reply:
Thank you! I was trying to figure out the easy syntax for this, but extracting it from the `map`ed example below was not working for me well.